### ISSUE TITLE
[Question] Question about "Refresh Interval" of the "Repeated refresh" mode in the "Auto Refresh mode

### ISSUE THREAD

**Issue by johntaxmi** [kodi-pvr/pvr.iptvsimple#618]
Is it possible to make the "Refresh Interval" of the "Repeated refresh" mode in the "Auto Refresh mode" even lower that 0 Minutes?
I wanna know if its possible to make that in seconds, like refresh every 15 seconds.

I am trying to force edit the settings.xml from the addon file but it does not seem that I can make that work.

**Comment by phunkyfish** [comment]
Why?

**Comment by johntaxmi** [comment]
I just wanna have my m3u file refresh quicker than that.
Is that a possible change that I can make, please?

**Comment by phunkyfish** [comment]
No, I’m afraid not. Supporting extra units of time doesn’t really make sense just for this use case. If you want reload on command you should try IPTV Merge

**Comment by johntaxmi** [comment]
Hmmm, I  do not really need a reload command, i just need the m3u file to get refreshed while watching it, so it can pre download the next chunk of video.

**Comment by johntaxmi** [comment]
But if you do not want to implement that on the public settings of the addon is it possible to guidance me where exactly can i add more units of time in the already created addon?

**Comment by phunkyfish** [comment]
Maybe explain what you really want to do. On your first reply you said I just want refresh to be faster. So I’ll ask again, why?

**Comment by johntaxmi** [comment]
I play low-duration m3u8 file which update itself with an other low-duration chunk of video only some seconds before the first chunk ends.

And beacuse of the low-duration chunk, the m3u8 file ends before it can get refreshed from your addon and the file just closes at KODI.

Thats why i want the refresh from the addon to be quicker so it can pre-load the next chunk of video from the m3u8 file before the first one ends.

**Comment by phunkyfish** [comment]
Ok, but an m3u8 video stream is designed to cater for this scenario. The question is have is why is yours different. It would appear your provider trying to use a VOD format for streaming channels.

Note that M3u8’s regularly have 0.1 to 1 second segments so I’m not sure 15 seconds should pose a problem.

**Comment by johntaxmi** [comment]
I am not really sure, i am trying to use greek international m3u8 links, and am 100% sure and confident that if i can lower that refresh rate to be in seconds the problem will not appear anymore.
I know that beacuse sometimes when the m3u8 actually refreshed before the first chunk ends then the video continues normally but other times when the refresh timer get a duration more than the total duration of the m3u8 chunk, the file just closes.

So thats exactly the reason i wanna lower the duration so i can always be inside the chunk duration.

Thats the reason i ask for your help to show me where and how to make it in seconds and no in minutes.

**Comment by phunkyfish** [comment]
What happens if you play it using inputstream.adaptive?

**Comment by johntaxmi** [comment]
Just tested, it was indeed a good idea.
But again it stops and the video closes after some seconds and i have to re open it to continue the live m3u8.

**Comment by phunkyfish** [comment]
Ok, please post a question on the adaptive GitHub, likely @glennguy will have an idea what is wrong with your stream.

**Comment by johntaxmi** [comment]
Returning to this, I did a small research, while doing it I try finding an other greek channel m3u8 link, and i found one that each chunk last more than 1 minute before closing.
So i play with your auto refresh mode.
And as I expected when i had the refresh mode at 0 minutes the m3u8 was playing normally but when i changed it up to 2 minutes the m3u8 was closing at around 1 minute.
So now I am not only 100% sure but i am also putting my hand on that if there is a way that i can make your refresh mode lower than 0 minutes(around 15 seconds) i will not continue to have this issue.

How should i question it to the other github page?

**Comment by phunkyfish** [comment]
Just because it could work with your workaround doesn’t mean it should.

Try posting here: https://github.com/xbmc/inputstream.adaptive/issues. Please give a sample m3u8

**Comment by johntaxmi** [comment]
Returing back to this, every how many seconds does the m3u8 file refresh while having it at 0 Minutes?
Is it a random amount of seconds between, 1 and 59? Or is it a fixed number?

**Comment by phunkyfish** [comment]
Not more often than every 2 seconds from looking at the code.

**Comment by johntaxmi** [comment]
Oh, so it does indeed refresh the m3u8 file every 2 seconds?
Is that fixed at 2 seconds or does that number change?

**Comment by phunkyfish** [comment]
It will vary depending on how many tasks the add-on has to perform each loop. I would not be relying on it being exact.

I have a feeling though that your m3u8 stream is the problem here. Do you have a example file to share?

**Comment by johntaxmi** [comment]
You can try this m3u8 for example: 
http://tvalb.xyz:8080/9rht8uKGJs/sBN1pWJUhQ/3201
Its OPEN BEYOND, an international greek channel.
https://www.tvopen.gr/live

**Comment by phunkyfish** [comment]
I can view the stream in a browser but I get a 404 if I try to download the M3U8 file.

**Comment by glennguy** [comment]
> You can try this m3u8 for example:
http://tvalb.xyz:8080/9rht8uKGJs/sBN1pWJUhQ/3201

Empty response for this link where I am...
I pulled this from one of the responses in the browser: https://liveopencloud.siliconweb.com/1/ZlRza2R6L2tFRnFJ/eWVLSlQx/hls/live/playlist.m3u8

Please see findings here: https://github.com/xbmc/inputstream.adaptive/issues/1024#issuecomment-1194018756

**Comment by phunkyfish** [comment]
@johntaxmi it appears @glennguy has arrived at the same conclusion I was trying to discern.

You must give a master playlist to iptvsimple. A sub/variant playlist will not ever work I’m afraid. You should not need to refresh other than to update the list of channels/epg data.

**Comment by johntaxmi** [comment]
That actually make sense.
So everytime that I want to watch a stream I have to find the master playlist.
But how can find the master each time?
I can not even find the link you mention above.
What is the way to find the master playlist from a site?
I always go to the Network and find an m3u8 file, is there any other way?
I mean like what is the main playlist at this channel(An other greek channel)
https://watch.antennaplus.gr/#/live/ant1
As the m3u8 files change every 3 seconds or so

Also an other question that i have is, why then my link worked normally and other times it was just stopping?
What could be that issue about?

**Comment by phunkyfish** [comment]
We can’t help you there. You’ll need to figure that out yourself. It can be different for every site/provider and they won’t necessarily make it easy to figure out.

**Comment by glennguy** [comment]
@johntaxmi to be very clear:
> You must give a master playlist to iptvsimple.

This is via the channel m3u8. Don't give iptvsimple a master or sub/variant m3u8 in the settings, only a m3u8 *containing* the url to a master m3u8.

**Comment by johntaxmi** [comment]
@glennguy You mean to create a m3u8 file that contain inside it the master m3u8 link?
